### ISSUE TITLE
GTK 3.20

### ISSUE THREAD

**Issue by ronasimi** [corebob/numix-solarized#1]
Are you planning to update this for GTK 3.20?

**Comment by corebob** [comment]
I don't have much time to work on this, so probably not.
Are there any issues with 3.20?

**Comment by ronasimi** [comment]
Yeah, it's pretty broken.

**Comment by oberon2007** [comment]
Yes. 3.20 brings massive changes and breaks all gtk3-themes. I would so much love to keep this theme, too! What was the original source you adapted to the solarized colors?

**Comment by corebob** [comment]
I make a package out of this one

http://bitterologist.deviantart.com/art/Numix-Solarized-417575928

The Light theme was ok, but the dark theme wasn't really the dark green solarized theme at all, so I replaced it with this

https://github.com/mkrnr/Numix-Solarized-Dark

so it's basically a mix of those

**Comment by oberon2007** [comment]
Thank you for the information!

**Comment by jhnsmth** [comment]
Hi! Sorry, I'm a bit late) I also had this issue. My workaround is to modify original numix theme with desired colours. [Here](https://gist.github.com/jhnsmth/6dfb0f55fe457feee8ccad62b1ca71a6) is a patch and script for solarized dark. And [here](http://imgur.com/sntc9SA) is the result. It's more closer to  the original theme made by  bitterologist on deviantart than to the one from this repo.
Updating and keeping it up this repo requires too much efforts,  so, I think, the easiest way to maintain the related arch package further will be to clone original numix from the git repo and apply patches during the build.

**Comment by oberon2007** [comment]
Thanks a lot! I will try your patch and script :smiley:

**Comment by oberon2007** [comment]
Could you explain how to compile the dark version?
I've tried making a package based on the [Archlinux-PKGBUILD](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/numix-themes) 
```
prepare() {
  cd numix-gtk-theme-${pkgver}
  cp gtk-3.0/assets/radio-{selected,checked}.svg
  cp gtk-3.0/assets/radio-{selected,checked}-dark.svg
  for FILE in gtk-3.0/assets/*.svg
  do
    sed -i 's/#d64937/#859900/Ig' "${FILE}"
    sed -i 's/#aaa/#eee8d6/Ig' "${FILE}"
    sed -i 's/#f9f9f9/#fdf6e3/Ig' "${FILE}"
    if [[ -f ${FILE%.svg}.png ]]
      then
      convert -resize 16x16 -background none "${FILE}" "${FILE%.svg}.png"
    fi
  done

  patch -p1 -i "../Solarization.patch"
}

build() {
  cd numix-gtk-theme-${pkgver}

  bundle install --path .
  make SASS="bundle exec sass"
}

package() {
  cd numix-gtk-theme-${pkgver}

  make SASS="bundle exec sass" DESTDIR="${pkgdir}" install
```
but that gives me only the light version...

**Comment by jhnsmth** [comment]
@oberon2007 actually, it's neither dark nor light. It's kinda a mixture of both. Panels,  drop-downs, menus are dark , while main area and text boxes are light.
To create a 'full' dark version you need to simply change the colours in the patch and in the script. As I remember, I couldn't just change the colours with a regex, like in the icons,  because I wanted to change the same colour to  the different ones.
This patch is generated from a commit in the local git repo, so it might be easier to locally clone numix repo, change the colours and make a patch, than modifying the patch file straight.
You need to  modify 3 files:
- gtk-2.0/gtkrc
- gtk-3.0/scss/_global.scss
- gtk-3.20/scss/_global.scss

The changes for gtk-3.20 and gtk-3.0 are pretty similar. For the colours values you can guide yourself by the corresponding files from this repo,  the names of the variables are almost the same. 
It would be nice to make a PKGBUILD for an arch package of the all 3 versions.
If you have any questions or need help feel free to ask.
